File: 프로그래머스/lv0/120861. 캐릭터의 좌표/캐릭터의 좌표.swift
import Foundation

func solution(_ keyinput:[String], _ board:[Int]) -> [Int] {
    var vertical = 0
    var horizontal = 0
    let verticalLimit = (board[1] - 1) / 2
    let horizontalLimit = (board[0] - 1) / 2

    for i in keyinput {
        if i == "up" && vertical < verticalLimit {
            vertical += 1
        }
        if i == "down" && vertical > -verticalLimit {
            vertical -= 1
        }
        if i == "left" && horizontal > -horizontalLimit {
            horizontal -= 1
        }
        if i == "right" && horizontal < horizontalLimit {
            horizontal += 1
        }
    }
    return [horizontal, vertical]
}

File: 프로그래머스/lv0/120861. 캐릭터의 좌표/캐릭터의 좌표.js
function solution(keyinput, board) {
    const firstMax = Math.floor(board[0] / 2);
    const secondMax = Math.floor(board[1] / 2);
    let first = 0;
    let second = 0;
    for (key of keyinput) {
        switch(key) {
            case "left":
                first = first === -firstMax ? -firstMax : first - 1;
                break;
            case "right":
                first = first === firstMax ? firstMax : first + 1;
                break;
            case "up":
                second = second === secondMax ? secondMax : second + 1;
                break;
            case "down":
                second =  second === -secondMax ? -secondMax : second - 1;
                break;
        }
    }
    return [first, second];
}